packages/ui/origin-ui-core/src/components/certificates/CertificatesInbox.tsx

```diff
@@ -0,0 +1,469 @@
+import React, { useEffect, useState } from 'react';
+import { useOriginConfiguration } from '../../utils/configuration';
+import { Button, Checkbox, Grid, Tab, Tabs, Typography } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    getCertificates,
+    ICertificateViewItem,
+    requestPublishForSale,
+    requestWithdrawCertificate
+} from '../../features/certificates';
+import { getEnvironment, getProducingDevices, getUserOffchain } from '../../features';
+import { getDeviceId } from '../../utils';
+import { IInboxCertificateData, IInboxItemData, InboxItem } from './Inbox/InboxItem';
+import { InboxSelectedItem } from './Inbox/InboxSelectedItem';
+import TextField from '@material-ui/core/TextField';
+import { BigNumber } from 'ethers';
+import { makeStyles } from '@material-ui/styles';
+
+export function CertificateInbox(): JSX.Element {
+    const configuration = useOriginConfiguration();
+    const { t } = useTranslation();
+
+    const certificates: ICertificateViewItem[] = useSelector(getCertificates);
+    const producingDevices = useSelector(getProducingDevices);
+    const environment = useSelector(getEnvironment);
+
+    const [allSelected, setAllSelected] = useState<boolean>(false);
```

Comment:
Where do you need this? 
Isn't there any better way to check if all the certificates are selected rather than creating another state object for this?

Reply:
it's used for "select all" checkbox, it required less checking of state of other certificates